server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -486,8 +486,18 @@ public ClusterState execute(ClusterState currentState) {
                                         // Remove all aliases - they shouldn't be restored
                                         indexMdBuilder.removeAllAliases();
                                     } else {
-                                        for (final String alias : snapshotIndexMetadata.getAliases().keySet()) {
-                                            aliases.add(alias);
+                                        for (final Map.Entry<String, AliasMetadata> alias : snapshotIndexMetadata.getAliases().entrySet()) {
+                                            String aliasName = alias.getKey();
+                                            if (request.renameAliasPattern() != null && request.renameAliasReplacement() != null) {
+                                                indexMdBuilder.removeAlias(aliasName);
+                                                aliasName = aliasName.replaceAll(
+                                                    request.renameAliasPattern(),
+                                                    request.renameAliasReplacement()
+                                                );
```

Comment:
Some thoughts here.

1. the regex pattern will be compiled every iteration through this loop.  It's better to compile the pattern outside the loop once (`Pattern pattern = Pattern.compile(request.renameAliasPattern());`) and then inside the loop do `pattern.matcher(aliasName).replaceAll(request.renameAliasReplacement)`
2. related to the above, `replace()` is preferred to `replaceAll` if the match is not a regex.  Is there any way to test that, or provide another possible configuration?  This may not be that important depending on how often this is executed.
3. will we always want to `replaceAll()` vs. `replaceFirst()`?

Reply:
1. Okay, done.
2. I don't see any trivial way to do that, and an index would have to have a LOT of aliases for this performance to matter.
3. This parameter pair is patterned off the existing rename_pattern/rename_replacement parameter pair which uses replaceAll. Though unlikely, there may be some use cases were replacing all is required, and I don't see any harm in using replace all.

Reply:
1. Awesome.
2. Agreed not worth the effort.
3. I'm fine with it if it matches the existing other parameter.